src/bootstrap/defaults/config.tools.toml

```diff
@@ -8,6 +8,8 @@ incremental = true
 download-rustc = "if-unchanged"
 
 [build]
+# cargo and clippy tests don't pass on stage 1
+test-stage = 2
```

Comment:
I don't think this is a good default -- it should be considered a bug if the tests do not work in stage 1. Certainly, they do work for Miri, so this can lead to Miri contributors waiting unnecessarily long for their builds to finish.

Reply:
https://github.com/rust-lang/rust/pull/137522 should be enough for the stage1 problems happen on clippy and cargo tests, so this should be revertable.

Reply:
So what that does is just bump the stage to 2? I guess that hides the bugs with those tools...

I have https://github.com/rust-lang/rust/issues/78717 somewhere in my TODO list, so I'll have to remember to revert that then. I don't know why cargo needs stage 2.

Reply:
I didn't get how https://github.com/rust-lang/rust/issues/78717 is being related to `test-stage = 2` in the config file.

> So what that does is just bump the stage to 2? I guess that hides the bugs with those tools...

Actually I don't know if it's a bug or not. All other tools work fine but cargo and clippy fail on stage 1 tests.

Reply:
https://github.com/rust-lang/rust/issues/78717 tracks the bug that you need stage 2 for clippy. Once that is fixed, clippy can be tested on stage 1.

Reply:
> Actually I don't know if it's a bug or not. 

I am arguing that we should consider it a bug. We should have the expectation that tools can be tested in stage 1 (except when they really need the bootstrap loop to have completed, which is quite rare).